src/components/accession2/withdraw/WithdrawModal.tsx

```diff
@@ -173,29 +192,42 @@ export default function WithdrawDialog(props: WithdrawDialogProps): JSX.Element
       setWithdrawAllSelected(false);
     }
 
+    if (isByWeight && accession.subsetCount && accession.subsetWeight) {
+      if (
+        accession.remainingQuantity?.units &&
+        accession.remainingQuantity?.units === 'Seeds' &&
+        accession.estimatedWeight?.units
+      ) {
+        estimated =
+          convertUnits(value, accession.estimatedWeight?.units, accession.subsetWeight.units) *
+          (accession.subsetCount / accession.subsetWeight.quantity);
```

Comment:
extract out as a helper function (within this context, maybe above line 196) 
```
  const calculateEstimated = (unit: string) => convertUnits(value, unit, accession.subsetWeight.units) * (accession.subsetCount / accession.subsetWeight.quantity);
```

Reply:
I like this idea, but even within the context of subsetWeight and subsetCount being defined Typescript thinks they might possibly be undefined within the function.

Reply:
My TypescriptFu is severly lacking though so maybe I'm missing the proper keyword to make this work.

Reply:
Share what you have? I can take a look.